mantidproject/mantid
# Use XML produced CTest over CxxTest/xmlrunner

**A note to gatekeepers**: The configuration of the Jenkins jobs needs to change after this is merged so that ctest unit parser is used.

**Description of work.**

Enables the CTest option `-T Test` in the Jenkins build scripts to force ctest to produce xUnit-compatible XML, which can be parsed by Jenkins. The main driver behind this is to have Jenkins capture better error reports when unit tests fail. The current XML capture mechanism misses certain scenarios such as early segfaults/crashes in C++ tests and early setup failures in Python unit tests. This *should* now capture all output from the separate process that CTest executes and stop confusion with developers having to dig through the console output.

**To test:**

* Code review.
* Clean the C++ unit tests and force a rebuild without `CCACHE` (CCACHE_DISABLE=1) to ensure the tests build properly.
* Run `ctest` as normal and check it passes for a selection of Python and C++ unit tests. Check that no new .xml files appear in `bin/Testing` (clear out the old ones first).
* Check Jenkins jobs have parsed the unit tests correctly so show they all passed. 

*There is no associated issue.*

*This does not require release notes* because **it is a developer-centric change.**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
